src/html/mod.rs

```diff
@@ -70,6 +72,60 @@ const FUSE_FILENAME: &str = "fuse.js";
 const SEARCH_JS: &str = include_str!("./templates/pages/search.js");
 const SEARCH_FILENAME: &str = "search.js";
 
+#[cfg(target_arch = "wasm32")]
+trait WasmToFilePath {
+  fn to_file_path(&self) -> Result<PathBuf, ()>;
+}
+
+#[cfg(target_arch = "wasm32")]
+impl WasmToFilePath for url::Url {
```

Comment:
Maybe use deno_path_util crate?

Reply:
oh wasnt aware it had that functionality built in. updated